src/util.rs

```diff
@@ -0,0 +1,18 @@
+use std::iter::Peekable;
+use std::slice::Iter;
+
+trait TakeWord {
+    fn take_word(&mut self) -> Option<Vec<u8>>;
+}
+
+impl TakeWord for Peekable<Iter<'_, u8>> {
```

Comment:
i don't really know what this `'_` is doing, but the error this gets from CI makes it look like i need it